### PR TITLE
HPCC-28661 Improve stripped build package naming

### PR DESCRIPTION
**Pull request by GordonSmith**
Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@jakesmith @richardkchapman  what do you think about this?  Will it be confusing for debug builds?  I assume we also need to default our builds to stripping symbols so that the correct rpm is built.  Is that going to cause us problems?
I think this naming convention is a good idea for docker images.  Not so sure about rpms, but it does make sense to be consistent.

**Comment by GordonSmith** [comment]
It is worth noting that we are currently stripping symbols for all "Release" builds...

**Comment by jakesmith** [comment]
It'll be okay for developers IMO, they will continue to be built without stripping, albeit with a package name change.
Release, as GS says, already stripped, and the default, so that package name will remain the same.
Agree it would also make sense for docker images to follow the same convention

File: CMakeLists.txt [hunk 1]
```diff
@@ -304,6 +304,10 @@ if(TOP_LEVEL_PROJECT)
         set(CPACK_STRIP_FILES TRUE)
     endif()
 
+    if (NOT CPACK_STRIP_FILES) 
+        set(packageStrippedLabel "_with_symbols")
```
Comment (GordonSmith) [inline CMakeLists.txt:308]: Rename to `_withsymbols`

**Comment by GordonSmith** [comment]
@ghalliday @jakesmith rebased onto newbuild and renamed to "_withsymbols" (ensure each `_` denotes a different thing)

**Review by ghalliday — APPROVED** [review]

**Comment by ghalliday** [comment]
@jakesmith @cloLN care you happy with this change?  I think it means all our rpms will change names.  What issues will that cause?

**Review by jakesmith — APPROVED** [review]
@GordonSmith - looks fine by me.

**Pull request merged**